connector/protobuf/src/test/scala/org/apache/spark/sql/protobuf/ProtobufFunctionsSuite.scala

```diff
@@ -1150,6 +1221,32 @@ class ProtobufFunctionsSuite extends QueryTest with SharedSparkSession with Prot
     }
   }
 
+  test("Retain empty recursive proto fields when retain.empty.message.types=true") {
+    // This verifies that a empty proto like 'message A { A a = 1}' can be retained by
+    // inserting a dummy field.
+
+    val emptyProtoSchema =
+      StructType(StructField("__dummy_field_in_empty_struct", StringType) :: Nil)
+    val expectedSchema = StructType(
```

Comment:
Could you add a comment here showing expected schema? It is hard to see here.

Reply:
Could you add comment to show this schema? It is hard to see.

Reply:
Thanks. The comment above looks good. Could you update the test with recursive depth set to 4?